collector/sysctl_linux.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	sysctlInclude     = kingpin.Flag("collector.sysctl.include", "Select sysctl metrics to include").Strings()
+	sysctlIncludeInfo = kingpin.Flag("collector.sysctl.include-info", "Select sysctl metrics to include as info metrics").Strings()
+
+	sysctlInfoDesc = prometheus.NewDesc(prometheus.BuildFQName(namespace, "sysctl", "info"), "sysctl info", []string{"name", "value", "index"}, nil)
+)
+
+type sysctlCollector struct {
+	fs      procfs.FS
+	logger  log.Logger
+	sysctls []*Sysctl
+}
+
+func init() {
+	registerCollector("sysctl", defaultEnabled, NewSysctlCollector)
```

Comment:
I'm not sure we want this enabled by default since it requires configuration.

We could auto-enable it if the include/include-info flags are passed. But leaving it `defaultDisabled` will avoid having it show up in the meta metrics for all users.